src/app/components/ThemeProvider/index.tsx

```diff
@@ -1,34 +1,76 @@
 import React, { PropsWithChildren } from 'react';
+import type { LoadableComponent } from '@loadable/component';
+import nodeLogger from '#lib/logger.node';
+import { THEME_PROVIDER_ERROR } from '#app/lib/logger.const';
 import defaultServiceVariants from '../../lib/config/services/defaultServiceVariants';
-import { Services, Variants } from '../../models/types/global';
+import {
+  ServicesVariantsProps,
+  Variants,
+  ServicesWithVariants,
+  ServicesWithNoVariants,
+} from '../../models/types/global';
 import themes from './themes/loadableConfig';
 import fallBackTheme from './themes/news';
 
-interface Props {
-  service: Services;
-  variant?: Variants | null;
-}
+const logger = nodeLogger(__filename);
+
+type ThemeComponentLoadable = LoadableComponent<{ children: React.ReactNode }>;
+type FallbackThemeComponent = React.FC<{ children: React.ReactNode }>;
+
+type NonVariantThemesType = {
+  [_service in ServicesWithNoVariants['service']]: ThemeComponentLoadable;
+};
+
+type VariantThemesType = {
+  [_service in ServicesWithVariants['service']]: {
+    [_variant in Variants]?: ThemeComponentLoadable;
+  };
+};
+
+const nonVariantThemes = Object.fromEntries(
+  Object.entries(themes).filter(([_service, theme]) =>
+    Object.keys(theme).includes('render'),
+  ),
+) as NonVariantThemesType;
+
+const variantThemes = Object.fromEntries(
+  Object.entries(themes).filter(
+    ([_service, theme]) => !Object.keys(theme).includes('render'),
+  ),
+) as VariantThemesType;
 
 export const ThemeProvider = ({
   children,
   service,
-  variant,
-}: PropsWithChildren<Props>) => {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let LoadableContextProvider: any = fallBackTheme;
+  ...rest
```

Comment:
Is there a reason why we've changed this to a spread operator instead of destructuring variant directly? TS weirdness?